spec/models/long_read_qc_result_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe LongReadQcResult, type: :model do
+  describe '#create' do
+    it 'errors without a labware_barcode' do
+      errored = false
+      begin
+        create :long_read_qc_result, labware_barcode: nil
+      rescue StandardError
+        errored = true
+      end
+      expect(errored).to eq(true)
+    end
+
+    it 'errors without a sample_id' do
+      errored = false
+      begin
+        create :long_read_qc_result, sample_id: nil
+      rescue StandardError
+        errored = true
+      end
+      expect(errored).to eq(true)
+    end
+
+    it 'errors without a assay_type' do
+      errored = false
+      begin
+        create :long_read_qc_result, assay_type: nil
+      rescue StandardError
+        errored = true
+      end
+      expect(errored).to eq(true)
+    end
+
+    it 'errors without a key' do
+      errored = false
+      begin
+        create :long_read_qc_result, key: nil
+      rescue StandardError
+        errored = true
+      end
+      expect(errored).to eq(true)
+    end
+
+    it 'errors without a value' do
+      errored = false
+      begin
+        create :long_read_qc_result, value: nil
+      rescue StandardError
+        errored = true
+      end
+      expect(errored).to eq(true)
+    end
+
+    describe 'units' do
+      it 'inserts with a value for units' do
+        qc_result = create :long_read_qc_result, units: 'unit1'
+        expect(qc_result.units).to be_truthy
+      end
+
+      it 'is nullable' do
+        qc_result = create :long_read_qc_result, units: nil
+        expect(qc_result.units).to be_nil
+      end
+    end
```

Comment:
I think this test won't be needed because it is functionality already provided by the factory.

I think the important in here is to test create_or_update_from_json. Have a look at https://github.com/sanger/unified_warehouse/blob/develop/spec/models/qc_result_spec.rb